src/daft-scan/src/lib.rs

```diff
@@ -481,47 +482,30 @@ impl ScanTask {
             sc1.schema.clone(),
             sc1.storage_config.clone(),
             sc1.pushdowns.clone(),
-            sc1.file_path_column.clone(),
+            sc1.generated_fields.clone(),
         ))
     }
 
     #[must_use]
     pub fn materialized_schema(&self) -> SchemaRef {
-        match (&self.pushdowns.columns, &self.file_path_column) {
-            (None, None) => self.schema.clone(),
-            (Some(columns), file_path_column_opt) => {
-                let filtered_fields = self
-                    .schema
-                    .fields
-                    .clone()
-                    .into_iter()
-                    .filter(|(name, _)| columns.contains(name));
-
-                let fields = match file_path_column_opt {
-                    Some(file_path_column) => filtered_fields
-                        .chain(std::iter::once((
-                            file_path_column.to_string(),
-                            Field::new(file_path_column.to_string(), DataType::Utf8),
-                        )))
-                        .collect(),
-                    None => filtered_fields.collect(),
-                };
-
-                Arc::new(Schema { fields })
-            }
-            (None, Some(file_path_column)) => Arc::new(Schema {
-                fields: self
-                    .schema
-                    .fields
-                    .clone()
-                    .into_iter()
-                    .chain(std::iter::once((
-                        file_path_column.to_string(),
-                        Field::new(file_path_column.to_string(), DataType::Utf8),
-                    )))
-                    .collect(),
-            }),
+        let mut fields = self.schema.fields.clone();
```

Comment:
We don't need to clone the fields if there's no generated fields or pushdown columns

Reply:
Sounds good, elided the extra clone in this case